core/src/main/java/cz/xtf/core/git/JGitResolver.java

```diff
@@ -0,0 +1,140 @@
+package cz.xtf.core.git;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
+import org.eclipse.jgit.transport.RemoteConfig;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Try to resolve repository remote URL and branch from .git directory
+ * <p>
+ * This method tries to match HEAD commit to remote references.
+ * If there is a match, the remote URL and branch are set.
+ * For attached HEAD this method could be simplified with jgit methods, but
+ * "universal" approach was chosen, as for example Jenkins git plugin creates a detached state.
+ * In case of multiple matches, upstream or origin (in this order) is preferred.
+ * </p>
+ */
+@Slf4j
+class JGitResolver implements GitResolver {
+    private static final String URL_TEMPLATE = "https://%s/%s/%s";
+    private static String reference;
+    private static String url;
+
+    /**
+     * Try to set repository ref and URL from HEAD commit
+     */
+    public JGitResolver() {
+        try {
+            resolveRepoFromHEAD();
+        } catch (IOException | URISyntaxException e) {
+            log.error("Failed to resolve repository from HEAD", e);
+            throw new RuntimeException("Failed to resolve repository from HEAD with error: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Try to set repository ref and URL from HEAD commit
+     */
+    private static void resolveRepoFromHEAD() throws IOException, URISyntaxException {
+        //look for a git repository recursively till system root folder
+        Repository repository = new FileRepositoryBuilder().findGitDir().build();
+
+        if (repository == null) {
+            log.debug("Failed to find a git repository");
```

Comment:
can be these log into error instead of debug.. I would like to see a message if something is wrong. now you have to do extra step to enable debug logging to see what is wrong.